src/Sulu/Bundle/MediaBundle/Media/ImageConverter/ImagineImageConverter.php

```diff
@@ -12,116 +12,172 @@
 namespace Sulu\Bundle\MediaBundle\Media\ImageConverter;
 
 use Imagine\Gd\Imagine as GdImagine;
+use Imagine\Image\ImageInterface;
 use Imagine\Image\Palette\RGB;
+use Imagine\Imagick\Imagine;
 use Imagine\Imagick\Imagine as ImagickImagine;
 use RuntimeException;
 use Sulu\Bundle\MediaBundle\Media\Exception\ImageProxyInvalidFormatOptionsException;
 use Sulu\Bundle\MediaBundle\Media\Exception\ImageProxyMediaNotFoundException;
 use Sulu\Bundle\MediaBundle\Media\Exception\InvalidFileTypeException;
-use Sulu\Bundle\MediaBundle\Media\ImageConverter\Command\Manager\ManagerInterface;
+use Sulu\Bundle\MediaBundle\Media\ImageConverter\Transformation\Manager\ManagerInterface;
+use Sulu\Bundle\MediaBundle\Media\ImageConverter\Transformation\TransformationInterface;
 
 /**
  * Sulu imagine converter for media.
  */
 class ImagineImageConverter implements ImageConverterInterface
 {
     /**
-     * @var GdImagine|ImagickImagine
-     */
-    protected $image;
-
-    /**
      * @var ManagerInterface
      */
-    protected $commandManager;
+    private $transformationManager;
 
     /**
-     * @param ManagerInterface $commandManager
+     * @var TransformationInterface
      */
-    public function __construct(ManagerInterface $commandManager)
-    {
-        $this->commandManager = $commandManager;
-    }
+    private $scaleTransformation;
```

Comment:
I am not sure if I would implement that as a `scaleTransformation`, haven't we talked about that? It should not be accessible as transformation in the transformation part of the XML, right? Then I would not implement it as a transformation, but integrate it in a tighter way.